modules/spo-applying-profiles.adoc

```diff
@@ -0,0 +1,135 @@
+// Module included in the following assemblies:
+//
+// * security/security_profiles_operator/spo-seccomp.adoc
+// * security/security_profiles_operator/spo-selinux.adoc
+
+ifeval::["{context}" == "spo-seccomp"]
+:seccomp:
+:type: seccomp
+:kind: SeccompProfile
+endif::[]
+ifeval::["{context}" == "spo-selinux"]
+:selinux:
+:type: SELinux
+:kind: SelinuxProfile
+endif::[]
+
+:_content-type: PROCEDURE
+[id="spo-applying-profiles_{context}"]
+= Applying {type} profiles to a pod
+
+Create a Pod to apply one of the created profiles.
+
+.Procedure
+
+ifdef::seccomp[]
+. Create a Pod object that defines a `securityContext`:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: Pod
+metadata:
+  name: test-pod
+spec:
+  securityContext:
+    seccompProfile:
+      type: Localhost
+      localhostProfile: operator/my-namespace/profile1.json
+  containers:
+    - name: test-container
+      image: nginx
+----
+
+. View the profile path of the `seccompProfile.localhostProfile` attribute by running the following command:
++
+[source,terminal]
+----
+$ oc -n my-namespace get seccompprofile profile1 --output wide
+----
++
+.Example output
+[source,terminal]
+----
+NAME       STATUS   AGE   SECCOMPPROFILE.LOCALHOSTPROFILE
+profile1   Active   14s   operator/my-namespace/profile1.json
+----
+
+. Apply the profile to a Deployment by running the following command:
++
+[source,terminal]
+----
+$ oc --namespace my-namespace patch deployment myapp --patch '{"spec": {"template": {"spec": {"securityContext": {"seccompProfile": {"type": "localhost", "localhostProfile": "'$(oc --namespace my-namespace get seccompprofile profile1 --output=jsonpath='{.status.seccompProfile\.localhostProfile}')'}}}}}}'
+----
++
+.Example output
+[source,terminal]
+----
+deployment.apps/myapp patched
+----
+
+.Verification
+
+* Confirm the profile was applied correctly by running the following command:
++
+[source,terminal]
+----
+$ oc -n my-namespace get deployment myapp --output=jsonpath='{.spec.template.spec.securityContext}' | \
+jq .
+{
+  "seccompProfile": {
+    "localhostProfile": "operator/my-namespace/profile1.json",
+    "type": "localhost"
+  }
+}
+----
+endif::[]
+ifdef::selinux[]
+. Obtain the SELinux profile usage string by running the following command:
++
+[source,terminal]
+----
+$ oc get selinuxprofile.security-profiles-operator.x-k8s.io/nginx-secure -nnginx-deploy -ojsonpath='{.status.usage}'
+----
++
+.Example output
+[source,terminal]
+----
+nginx-secure_nginx-deploy.process%
+----
+
+. Apply the output string in the workload manifest in the `.spec.containers[].securityContext.seLinuxOptions` attribute:
```

Comment:
So I literally found out just now, but I think we need to document that the namespace must be labelled to allow the `privileged` Pod Security Admission Standard (see https://kubernetes.io/docs/concepts/security/pod-security-standards/).

This only applies to SELinux, for some reason, the `restricted` PSA allows you to set any seccomp profile just fine (and really any, including one that allows all syscalls), but in order to set a more secure SELinux profile, the namespace must be labeled to allow privileged workloads. I think that's kind of hilarious, but whatever.

Anyway, can we add a blob that says that with OCP 4.12, in order to satisfy the Pod Security Admission controller, the namespace must be labelled like this:
```
oc label ns nginx-deploy security.openshift.io/scc.podSecurityLabelSync=false
oc label ns nginx-deploy --overwrite=true pod-security.kubernetes.io/enforce=privileged
```

Reply:
`Privileged | Unrestricted policy, providing the widest possible level of permissions.`

Yeah that's hilarious. :joy:

Reply:
Could merge the two label commands into one:
`oc label ns nginx-deploy security.openshift.io/scc.podSecurityLabelSync=false pod-security.kubernetes.io/enforce=privileged --overwrite=true`
But above is also fine